### ISSUE TITLE
Sync executable with formal spec - CHAIN STS

### ISSUE THREAD

**Issue by uroboros** [input-output-hk/cardano-ledger-specs#748]
<img width="873" alt="Screenshot_FIG71" src="https://user-images.githubusercontent.com/8812/62703423-78fee500-b9e9-11e9-8021-54423af42767.png">

In the formal spec `ChainState` we have `h ∈ HashHeader` and in the exec. spec, `h ∈ HashHeader?`.

**Comment by mgudemann** [comment]
this is from https://github.com/input-output-hk/cardano-ledger-specs/pull/639/files
there is still the question about how the last Byron block and first Shelley block will be handled
The change was required in the exec spec and should be integrated into the formal spec

**Comment by JaredCorduan** [comment]
Yes, we should be consistent. @dcoutts suggested we could possibly use a sum type here to distinguish pointing to a Byron block or a Shelly block.
